grammar.y

```diff
@@ -1402,7 +1402,7 @@ insert_stmt:
     }
 
     if sel, ok := $5.(*Select); ok {
-      if sel.Having != nil || sel.GroupBy != nil {
```

Comment:
we're inside an `INSERT` with `SELECT` rule (see L1385). we're intentionally not allowing group by on those queries because we wanted to learn the use cases for it. but now we have one and we decided to remove the check